wizard_agent/brain/smart_collection_strategy.py

```diff
@@ -0,0 +1,119 @@
+from typing import List
+import operator
+from . import strategy
+from .utils import util_functions as utils, constants
+
+'''
+Strategy: 
+
+Retreive location of ammo and treasure blocks, 
+find the tile with highest score, and get 
+corresponding ideal path.
+
+'''
+
+
+class SmartCollectionStrategy(strategy.Strategy):
+    def __init__(self):
+    	self.ammo_state = {}
+    	self.prev_bombs = []
+
+        self.game_state = None
+        self.player_state = None
+
+        # value between 0-1, closer to 1 means it's more likely to go to tile
+        self.ammo_priority = 0.5
+        self.treasure_priority = 1
+        self.reachable_priority = 1
+        self.urgent_priority = 1
+
+    def execute(self, game_state: object, player_state: object) -> List[str]:
+        self.game_state = game_state
+        self.player_state = player_state
+        
+        location = player_state.location
+        ammo_blocks = game_state.ammo
+        treasure_blocks = game_state.treasure
+        bombs = game_state.bombs
+
+        exploded_bombs = self.get_exploded_bombs(bombs)
+        self.prev_bombs = bombs
+
+        empty_near_ammo = self.get_empty_near_blocks(ammo_blocks)
+        empty_near_treasure = self.get_empty_near_blocks(treasure_blocks)
+        all_empty = empty_near_ammo + empty_near_treasure
+ 
+
+        # retrieve the best tile
+        ideal_tile = self.get_ideal_tile(all_empty, empty_near_ammo, empty_near_treasure, location)
+
+        # navigate to ideal tile
+        if ideal_tile is not None:
+            path = utils.get_shortest_path(location, ideal_tile, game_state)
+            action_seq = utils.get_path_action_seq(location, path)
+            return action_seq
+
+        return [constants.ACTIONS["none"]]
+    
+    def can_execute(self, game_state: object, player_state: object) -> bool:
+        self.game_state = game_state
+        self.player_state = player_state
+
+        location = player_state.location
+        ammo = player_state.ammo
+        ammo_blocks = game_state.ammo
+        treasure_blocks = game_state.treasure
+        empty_near_ammo = self.get_empty_near_blocks(ammo_blocks)
+        empty_near_treasure = self.get_empty_near_blocks(treasure_blocks)
+        all_empty = empty_near_ammo + empty_near_treasure
+        reachable_tiles = utils.get_reachable_tiles(location, all_empty, game_state)
+        return ammo > 0 and reachable_tiles
```

Comment:
incorrect can_execute return condition. Shouldn't this strategy execute if ammo is less than the ammo capacity, and there are reachable ammo and treasure blocks?